spec/token_spec.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+RSpec.describe AppleSignIn::Token do
+  subject(:token_service) { described_class.new(code, grant_type, refresh_token) }
+
+  context '#authenticater' do
+    context 'when paramters are valid' do
+      let(:code) { 'valid_code' }
+      let(:grant_type) { 'valid_type' }
+      let(:refresh_token) { 'valid_token' }
+
+      before do
+        stub_request(:post, 'http://appleid.apple.com:443/auth/token')
+          .to_return(
+            body: {
+            }.to_json,
+            status: 200,
+            headers: { 'Content-Type': 'application/json' }
+          )
+          AppleSignIn.config.apple_client_id = 'client_id'
+          AppleSignIn.config.apple_private_key = OpenSSL::PKey::EC.new('prime256v1').generate_key!
+          AppleSignIn.config.apple_key_id = 'apple_kid'
+          AppleSignIn.config.apple_team_id = 'team_id'
+          AppleSignIn.config.redirect_uri = 'www.example.com'
```

Comment:
I think that writing it using the config block might give a better sense of scope
```
AppleSignIn.config do |config|
   config.apple_client_id = 'client_id'
  ...
end
```